src/widget/wsearchlineedit.h

```diff
@@ -70,22 +62,33 @@ class WSearchLineEdit : public QLineEdit, public WBaseWidget {
     // configuration value changes.
     static int s_debouncingTimeoutMillis;
 
-    void showPlaceholder();
-    void showSearchText(const QString& text);
-    void updateEditBox(const QString& text);
+    void refreshState();
 
+    void enableSearch(const QString& text);
+    void updateEditBox(const QString& text);
     void updateClearButton(const QString& text);
 
     QString getSearchText() const;
 
+    // Update the displayed text without (re-)starting the timer
+    void setTextBlockSignals(const QString& text);
+
     QToolButton* const m_clearButton;
 
     int m_frameWidth;
     int m_innerHeight;
 
-    QColor m_foregroundColor;
-
     QTimer m_debouncingTimer;
 
+    enum class State {
+        Active,
+        Inactive,
+    };
     State m_state;
```

Comment:
I'm unclear why this state exists at all. How is m_state different from QWidget::isEnabled? The only place it is checked now is WSearchLineEdit::updateClearButton, where I think QString::isEmpty may be sufficient. Deleting that check in updateClearButton doesn't seem to cause any problems from my brief testing, but maybe I am overlooking something.

Reply:
I'll give it a try to simplify this even further.

Reply:
It worked!!

Reply:
Hooray for less state!